src/SequenceManager.cc

```diff
@@ -358,4 +379,57 @@ void SequenceManager::store_sequence(const Sequence& seq) const
         store_step(seq_path / extract_filename_step(++idx, max_digits, step), step);
 }
 
+
+std::string SequenceManager::stage_files_in_directory(const std::string& glob1, const std::string& glob2)
+{
+    git_repo_.add(glob1);
+    git_repo_.update(glob1);
+    if (not glob2.empty()) {
+        git_repo_.add(glob2);
+        git_repo_.update(glob2);
+    }
+
+    auto git_msg = ""s;
+    for(const auto& elm: git_repo_.status()) {
+        if (elm.handling != "staged")
+            continue;
+        auto filename = std::filesystem::path{ elm.path_name }.filename();
+        git_msg = gul14::cat(git_msg, "\n- ", elm.changes, ": ", filename.c_str());
+    }
+    return git_msg;
+}
+
+std::string escape_glob(const std::string& path)
+{
+    auto escaped = ""s;
+    escaped.reserve(path.length());
+
+    for (const signed char c : path) {
```

Comment:
I don't remember where that piece of code comes from, but the `static_assert()` tells me that there was a specific reason for using `signed char`. I was just wondering if there is one here as well – but that does not seem to be the case.

Reply:
I have changed that from `signed char` to `char` as a drive-by change in the last commit.

Reply:
Thanks.

I believe I figured out what the problem was, ... the comparsion op

It is hard to write it if you do not know the signedness of c, because you probably get warnings for the bordering case if you expect unisgned (i.e. c > 127) and the type is in fact signed.

```c++
         default:
                if (c < 32)
                {
                    // This applies also to all non-ASCII characters (>127) because they
                    // are mapped to negative values
                    escaped += "\\x";
                    escaped += hex_string(c);
                }
```